### PR TITLE
para contract fixes

### PR DESCRIPTION
**Pull request by robert-e-davidson3**
Fixes for these tickets:
* https://github.com/AugurProject/augur/issues/9099
* https://github.com/AugurProject/augur/issues/9147
* https://github.com/AugurProject/augur/issues/9107 (I am NOT sure if I did this right)
* https://github.com/AugurProject/augur/issues/9109
* https://github.com/AugurProject/augur/issues/9110
* https://github.com/AugurProject/augur/issues/9140
* https://github.com/AugurProject/augur/issues/9148

### CODE REVIEW

File: packages/augur-core/src/contracts/para/ParaShareToken.sol [hunk 1]
```diff
@@ -108,12 +108,12 @@ contract ParaShareToken is ITyped, Initializable, ERC1155, ReentrancyGuard {
...
 
     function buyCompleteSetsInternal(IMarket _market, address _account, uint256 _amount) internal returns (bool) {
-        if (!isMarketInitialized(_market)) {
+        uint256 _numTicks = markets[address(_market)].numTicks;
+        if (_numTicks == 0) { // !isMarketInitialized(_market)
             initializeMarket(_market);
         }
 
         uint256 _numOutcomes = markets[address(_market)].numOutcomes;
-        uint256 _numTicks = markets[address(_market)].numTicks;
```
Comment (nuevoalex) [inline packages/augur-core/src/contracts/para/ParaShareToken.sol:116]: If the market isn't initialized and this is called wouldn't numTicks be 0 then at this point? I think it needs to be set again in the numTicks == 0 conditional block.

File: packages/augur-core/src/contracts/para/ParaShareToken.sol [hunk 2]
```diff
@@ -154,19 +154,19 @@ contract ParaShareToken is ITyped, Initializable, ERC1155, ReentrancyGuard {
      * @return Bool True
      */
     function buyCompleteSetsForTrade(IMarket _market, uint256 _amount, uint256 _longOutcome, address _longRecipient, address _shortRecipient) external returns (bool) {
-        if (!isMarketInitialized(_market)) {
+        uint256 _numTicks = markets[address(_market)].numTicks;
+        if (_numTicks == 0) { // !isMarketInitialized(_market)
```
Comment (nuevoalex) [inline packages/augur-core/src/contracts/para/ParaShareToken.sol:159]: Same issue here I think

File: packages/augur-core/src/contracts/para/ParaShareToken.sol [hunk 3]
```diff
@@ -252,25 +250,25 @@ contract ParaShareToken is ITyped, Initializable, ERC1155, ReentrancyGuard {
...
-
-        return (_creatorFee, _reportingFee);
     }
 
     function burnCompleteSets(IMarket _market, address _account, uint256 _amount, address _sourceAccount, bytes32 _fingerprint) private returns (uint256 _payout, uint256 _creatorFee, uint256 _reportingFee) {
         _fingerprint;
-        if (!isMarketInitialized(_market)) {
+
+        uint256 _numTicks = markets[address(_market)].numTicks;
+        if (_numTicks == 0) { // !isMarketInitialized(_market)
```
Comment (nuevoalex) [inline packages/augur-core/src/contracts/para/ParaShareToken.sol:269]: Same here

**Review by nuevoalex — APPROVED** [review]
I think the numTicks removals need to be resolved unless I'm missing something, otherwise looks good.

File: packages/augur-core/src/contracts/para/ParaShareToken.sol (hunk 1 above)
Comment (robert-e-davidson3) [reply packages/augur-core/src/contracts/para/ParaShareToken.sol:116]: ARGH right!

**Comment by robert-e-davidson3** [comment]
The test failures are pre-existing on para_deploys. So I manually rebased into para_deploys. Closing.

**Pull request closed without merging**